docs_nnx/guides/linen_to_nnx.rst

```diff
@@ -109,28 +109,41 @@ If you want to access NNX model parameters in the stateless, dictionary-like fas
   model = Model(784, 256, 10, rngs=nnx.Rngs(0))
 
 
-  # parameters were already initialized during model instantiation
+  # Parameters were already initialized during model instantiation.
 
   assert model.linear.bias.value.shape == (10,)
   assert model.block.linear.kernel.value.shape == (784, 256)
 
 
-Training Step and Compilation
-==========
+Training step and compilation
+=============================
+
+Now, let’s proceed to writing a training step and compiling it using `JAX just-in-time compilation <https://jax.readthedocs.io/en/latest/jit-compilation.html>`__. Below are certain differences between Flax Linen and Flax NNX approaches.
+
+Compiling the training step:
+
+* Flax Linen uses ``@jax.jit`` - a `JAX transform <https://jax.readthedocs.io/en/latest/key-concepts.html#transformations>`__ - to compile the training step.
+* Flax NNX uses :meth:`@nnx.jit<flax.nnx.jit` - a `Flax NNX transform <https://flax.readthedocs.io/en/latest/guides/jax_and_nnx_transforms.html>`__ (one of several transform APIs that behave similarly to JAX transforms, but also `work well with Flax NNX objects <https://flax.readthedocs.io/en/latest/guides/jax_and_nnx_transforms.html>`__). So, while ``jax.jit`` only accepts pure stateless arguments, ``nnx.jit`` allows the arguments to be stateful NNX Modules. This greatly reduced the number of lines needed for a train step.
```

Comment:
More nits when I reread this: 
"compile the training step" -> "compile the training step, which has to be a pure function"
"only accepts pure stateless arguments" -> "only accepts functions pure stateless arguments"

Reply:
Good catch! Thanks